go/vt/tabletserver/sqlquery.go

```diff
@@ -239,14 +239,16 @@ func (sq *SqlQuery) Begin(context context.Context, session *proto.Session, txInf
 		return NewTabletError(ErrRetry, "Invalid session Id %v", session.SessionId)
 	}
 	defer queryStats.Record("BEGIN", time.Now())
-	txInfo.TransactionId = sq.qe.txPool.Begin()
+	ctx, cancel := withTimeout(ctx, sq.qe.txPool.poolTimeout.Get())
```

Comment:
I think we should go the other way round. Move this timeout (as well as the query timeout) into SqlQuery. Nobody else should be looking at those values anyways.